### ISSUE TITLE
(cdk synth): incorrect and confused message about `spawnSync docker ENOENT`

### ISSUE THREAD

**Issue by unchaptered** [aws/aws-cdk#22997]
### About incorrect and confused error meesage in aws-cdk
---

I think, **spawnSync docker ENOENT** is very confused message to some developer.

When I use `aws synth`, `aws destory` and `aws deploy`, I encounter that error.
IT feels **HEY! Your docker or docker image have some error!** to developer, using AWS CDK LIB(node.js).

And when search about this error, I can't find any solution to me, except DOCKER in aws/aws-cdk repo.

- #8707
- #9348

**But, actually this error is occured by esbuild is not installed in local computer.**

After install esbuild in project, I should fix this error.
So, I think aws-cdk changes this error message for all node.js developer, please.

### Links

MY clone project and issue.

- project https://github.com/unchaptered/e-commerce-be
- issue https://github.com/unchaptered/e-commerce-be/issues/1

**Comment by indrora** [comment]
ESBuild on windows is a tricky situation. @rix0rrr do we have a full build-and-run guide for windows yet?

**Comment by unchaptered** [comment]
> ESBuild on windows is a tricky situation. @rix0rrr do we have a full build-and-run guide for windows yet?

Oh, sorry to late response to @indrora.

I think I explained it to make u feels confused.

Yes, clearly I use windows OS.
But, when I solved this error, I find the references it with MacOS.

So, I guess...
When someone run cdk-cli without ESBuild, many peoeple could see this situation, I guess...

**Comment by rix0rrr** [comment]
Yes, catching the `ENOENT` error and throwing a descriptive message in its stead (that explains some alternatives) would be a good idea.

**Comment by wilhen01** [comment]
+1 for this - just encountered it on OS X and the issue was actually that I was using `NodejsFunction` and handed included `esbuild` as a project dependency. 

The docs for `NodejsFunction` could probably also be improved to indicate what peer-dependencies are required to make it work.

**Comment by dineshtrivedi** [comment]
+1 for this. Adding a comment to subscript to this issue.

Having a better error message would be great. I am getting `spawnSync docker ENOENT` error when trying to build a python lambda from within a container in Jenkins. The same situation as [here](https://github.com/aws/aws-cdk/issues/9348#issuecomment-678430129), but at this point I cannot use the same solution as mentioned there.

**Comment by dineshtrivedi** [comment]
@IsmaelMartinez , I am building the docker image as previous step and pushing to ECR. The cdk code uses the docker image from ECR repository.
